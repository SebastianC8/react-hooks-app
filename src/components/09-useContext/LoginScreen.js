import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    const { setUser } = useContext(UserContext);

    const formatArray = {
        id : 987,
        name: 'Anonymous'
    };

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>
            <button className="btn btn-primary" onClick = { () => setUser(formatArray) }>Asignar</button>
        </div>
    )
}
