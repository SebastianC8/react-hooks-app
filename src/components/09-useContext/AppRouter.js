import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
  Link
} from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar/>
                <div className="container mt-4">
                    <Switch>
                        <Route exact path="/" component = { HomeScreen }/>
                        <Route exact path="/about" component = { AboutScreen }/>
                        <Route exact path="/login" component = { LoginScreen }/>
                        <Redirect to="/"></Redirect>
                    </Switch>
                </div>
            </div>
        </Router>
    )
}
