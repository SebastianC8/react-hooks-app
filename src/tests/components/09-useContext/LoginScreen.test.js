import React from 'react';
import { mount } from 'enzyme';
import { UserContext } from '../../../components/09-useContext/UserContext';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';

describe('Pruebas en <LoginScreen/>', () => {
   
    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value = {{ setUser }}>
            <LoginScreen/>
        </UserContext.Provider>
    );

    test('should show ok', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('debe ejecutar setUser con el argumento esperado', () => {
    
        wrapper.find('button').prop('onClick')();

        expect(setUser).toHaveBeenCalledWith({
            id : 987,
            name: 'Anonymous'
        });

    });
    
});

