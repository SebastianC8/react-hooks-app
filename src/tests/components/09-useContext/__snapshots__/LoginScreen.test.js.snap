// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen/> should show ok 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Asignar
  </button>
</div>
`;
